packages/odata-v2/src/de-serializers/temporal-de-serializers.ts

```diff
@@ -0,0 +1,135 @@
+    serialize: serializeFromPlainDateTime,
+    serializeToUri: (value, serialize) =>
+      `datetime'${serialize(value).toString().replace(/Z$/, '')}'`
+  } as DeSerializer<Temporal.PlainDateTime>,
+  'Edm.DateTimeOffset': {
+    deserialize: deserializeToZonedDateTime,
+    serialize: serializeFromZonedDateTime,
+    serializeToUri: (value, serialize) => `datetimeoffset'${serialize(value)}'`
+  } as DeSerializer<Temporal.ZonedDateTime>,
+  'Edm.Time': {
+    deserialize: deserializeToPlainTime,
+    serialize: serializeFromPlainTime,
+    serializeToUri: (value, serialize) => `time'${serialize(value)}'`
+  } as DeSerializer<Temporal.PlainTime>
+};
+
+/**
+ * @internal
+ */
+export function deserializeToPlainDateTime(
+  edmDateTime: string
+): Temporal.PlainDateTime {
+  const dateTimeOffsetComponents = parseEdmDateTimeOffset(edmDateTime);
+  const dateTimeInstant = Temporal.Instant.fromEpochMilliseconds(
+    parseInt(dateTimeOffsetComponents.ticks)
+  );
+  return Temporal.PlainDateTime.from(dateTimeInstant.toString());
+}
+
+/**
+ * @internal
+ */
+export function deserializeToZonedDateTime(
+  edmDateTime: string
+): Temporal.ZonedDateTime {
+  const dateTimeOffsetComponents = parseEdmDateTimeOffset(edmDateTime);
+  const dateTimeInstant = Temporal.Instant.fromEpochMilliseconds(
+    parseInt(dateTimeOffsetComponents.ticks)
+  );
+  if (dateTimeOffsetComponents.sign && dateTimeOffsetComponents.offset) {
+    const timeZone = Temporal.TimeZone.from(
+      dateTimeOffsetComponents.sign + dateTimeOffsetComponents.offset
+    );
+    return dateTimeInstant.toZonedDateTimeISO(timeZone);
+  }
+  return dateTimeInstant.toZonedDateTimeISO('UTC');
+}
+
+function parseEdmDateTimeOffset(edmDateTime: string): {
+  [key: string]: string;
+} {
+  const dateTimeOffsetComponents =
+    /^\/Date\((?<ticks>\d+)((?<sign>[+-])(?<offset>\d{4}))?\)\/$/.exec(
+      edmDateTime
+    )?.groups;
+  if (!dateTimeOffsetComponents) {
+    throw new Error(`Failed to parse edmDateTime: ${edmDateTime}.`);
+  }
+  return dateTimeOffsetComponents;
+}
+
+/**
+ * @internal
+ */
+export function serializeFromPlainDateTime(
+  value: Temporal.PlainDateTime
+): string {
+  const instant = Temporal.TimeZone.from('UTC').getInstantFor(value);
+  return `/Date(${instant.epochMilliseconds})/`;
+}
+
+/**
+ * @internal
+ */
+export function serializeFromZonedDateTime(
+  value: Temporal.ZonedDateTime
+): string {
+  const operator = value.offset[0];
+  const offset = value.offset.replace(/[-+:]/g, '');
+  return `/Date(${value.epochMilliseconds}${operator}${leftpad(offset, 4)})/`;
+}
+
+/**
+ * @internal
+ */
+export function deserializeToPlainTime(value: string): Temporal.PlainTime {
+  const regexResult =
+    /PT(?<hours>\d{1,2}H)?(?<minutes>\d{1,2}M)?(?<seconds>\d{1,2}S)?/.exec(
+      value
+    );
+  if (!regexResult) {
+    throw new Error(`Failed to parse the value: ${value} to time.`);
+  }
+  const { hours, minutes, seconds } = regexResult?.groups || {};
```

Comment:
I know that this is very compact and handles the three groups. However, if a value is not there it is `undefined` and the parseInt(undefined) gives then a `NaN` which seems to be lead to 0 in temporal. I know there is a test, but I would prefer to initialize the values with "0" if they are not there so that the parsing and passing to the new Temporal.PlainTime constructor is not critical.